tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_NormalVectorQuantities.cpp

```diff
@@ -0,0 +1,298 @@
+                   Dim>>(*box)
+                   .at(direction));
+      CHECK_ITERABLE_APPROX(
+          (DataVector{fields_on_face.number_of_grid_points(), 1.0}),
+          sqrt(get(
+              dot_product(normal_covector, normal_covector,
+                          get<InverseSpatialMetric<Dim>>(fields_on_face)))));
+    }
+
+    for (const double t :
+         get(get<evolution::dg::Tags::InternalFace::MagnitudeOfNormal>(
+             *get<evolution::dg::Tags::InternalFace::NormalCovectorQuantities<
+                  Dim>>(*box)
+                  .at(direction)))) {
+      CHECK(t > 0.0);
+    }
+    for (size_t i = 0; i < Dim; ++i) {
+      CHECK_ITERABLE_APPROX(
+          unnormalized_normal_covectors.at(direction).get(i),
+          DataVector{
+              get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+                  *get<evolution::dg::Tags::InternalFace::
+                           NormalCovectorQuantities<Dim>>(*box)
+                       .at(direction))
+                  .get(i) *
+              get(get<evolution::dg::Tags::InternalFace::MagnitudeOfNormal>(
+                  *get<evolution::dg::Tags::InternalFace::
+                           NormalCovectorQuantities<Dim>>(*box)
+                       .at(direction)))});
+    }
+  }
+}
+
+template <size_t Dim>
+void test_flat_space(const bool use_moving_mesh) {
+  INFO("Testing flat space");
+  CAPTURE(Dim);
+  CAPTURE(use_moving_mesh);
+  const size_t number_of_grid_points_per_dim = 3;
+  auto box = create_box<Dim>(number_of_grid_points_per_dim, use_moving_mesh);
+  check_normal_covector_quantities<true, Dim>(make_not_null(&box));
+
+  if (use_moving_mesh) {
+    // Mutate the x component of the unnormalized normal vector to simulate
+    // moving mesh
+    db::mutate<ScaleZeroIndex>(
+        make_not_null(&box),
+        [](const gsl::not_null<double*> scale_zero_index) noexcept {
+          *scale_zero_index = 2.3;
+        });
+
+    check_normal_covector_quantities<true, Dim>(make_not_null(&box));
+  }
+}
+
+template <size_t Dim>
+void test_curved_space(const bool use_moving_mesh) {
+  INFO("Testing curved space");
+  CAPTURE(Dim);
+  CAPTURE(use_moving_mesh);
+  const size_t number_of_grid_points_per_dim = 3;
+  auto box = create_box<Dim>(number_of_grid_points_per_dim, use_moving_mesh);
+  check_normal_covector_quantities<false, Dim>(make_not_null(&box));
+
+  if (use_moving_mesh) {
+    // Mutate the x component of the unnormalized normal vector to simulate
+    // moving mesh
+    db::mutate<ScaleZeroIndex>(
+        make_not_null(&box),
+        [](const gsl::not_null<double*> scale_zero_index) noexcept {
+          *scale_zero_index = 2.3;
+        });
+
+    check_normal_covector_quantities<false, Dim>(make_not_null(&box));
+  }
+}
+
+SPECTRE_TEST_CASE("Unit.Evolution.DG.NormalVectorQuantities",
+                  "[Unit][Evolution][Actions]") {
+  for (const auto use_moving_mesh : {true, false}) {
```

Comment:
Maybe I'm being nit-picky here, but since you know the values of use_moving_mesh at compile time, isn't there a way to do the loop over use_moving_mesh at compile time?

Reply:
It's much easier to loop over things at runtime than compile time, and since the actual code in the evolution determines whether the mesh is moving at runtime. Mostly, reduces the number of  templates to instantiate :shrug: I'd agree with you in the production part of the code, here I'm going for "easier to read"